pkg/types/installconfig.go

```diff
@@ -154,6 +154,9 @@ type InstallConfig struct {
 	// +optional
 	Publish PublishingStrategy `json:"publish,omitempty"`
 
+	// InternalResources controls the visibility of ingress and apiserver. Defaults to public.
+	InternalResources *InternalResources `json:"internal,omitempty"`
```

Comment:
We should brainstorm a bit on the naming here `InternalResources` and `internal` are a bit vague. I think the effect here is we are distinguishing the publishing strategy for individual cluster components, so we could do something like `OperatorPublishingStrategy`, which would be distinguished from the previous publishing strategy which was for the cluster as a whole.